scripts/Muon/GUI/Common/contexts/fitting_contexts/tf_asymmetry_fitting_context.py

```diff
@@ -13,10 +13,68 @@ class TFAsymmetryFittingContext(GeneralFittingContext):
     def __init__(self, allow_double_pulse_fitting: bool = False):
         super(TFAsymmetryFittingContext, self).__init__(allow_double_pulse_fitting)
 
+        # A list of FitInformation's detailing all the TF Asymmetry fits that have happened including the fits that have
+        # been overridden by an updated fit. The last TF Asymmetry fit performed is at the end of the list, and undoing
+        # will remove it.
+        self._tf_asymmetry_single_fits_history: list = []
+        self._tf_asymmetry_simultaneous_fits_history: list = []
+
         self._tf_asymmetry_mode: bool = False
+
         self._tf_asymmetry_single_functions: list = []
         self._tf_asymmetry_simultaneous_function: IFunction = None
 
+        self._normalisations_for_undo: list = []
+        self._normalisations_fixed_for_undo: list = []
+
+    def all_latest_fits(self):
+        """Returns the latest unique fits for all fitting modes."""
+        latest_tf_single_fits = self._latest_unique_fits_in(self._tf_asymmetry_single_fits_history)
+        latest_tf_simultaneous_fits = self._latest_unique_fits_in(self._tf_asymmetry_simultaneous_fits_history)
+        return super().all_latest_fits() + latest_tf_single_fits + latest_tf_simultaneous_fits
+
+    @property
+    def active_fit_history(self):
+        """Returns the fit history for the currently active fitting mode."""
+        if self.tf_asymmetry_mode:
+            return self._tf_asymmetry_simultaneous_fits_history if self.simultaneous_fitting_mode else \
```

Comment:
would it be cleaner to have a single object for the single and simultaneous histories? You could use a dict with a single and sim keys to keep them separate.

Reply:
That is a lot better, thanks!